odoo/addons/base/static/src/js/res_config_settings.js

```diff
@@ -0,0 +1,268 @@
+        this._super.apply(this, arguments);
+
+        var self = this;
+        this.modules = [];
+
+        _.each(this.$('.setting_view'),function(settingView, index) {
+            var group = !$(settingView).hasClass('o_invisible_modifier');
+            if(group) {
+                var string = $(settingView).attr('data-string');
+                var key = $(settingView).attr('data-key');
+                var imgurl = self._getAppIconUrl(key);
+                var view = $(settingView);
+                self.modules.push({
+                    key: key,
+                    string: string,
+                    settingView: view,
+                    imgurl: imgurl,
+                    order: key=== self.activeSettingTab ? 0 : index+1
+                });
+                if (config.isMobile) {
+                    self._enableSwipeOnSettingView(view);
+                }
+                view.addClass("o_hidden");
+                view.prepend($("<div>").html('<img class="icon" src="'+imgurl+'"><span class="appName">'+string+'</span>').addClass('settingSearchHeader o_hidden'));
+            } else {
+                $(settingView).remove();
+            }
+        });
+
+        this.modules = _.sortBy(this.modules,function(m){return m.order});
+        var tabs = $(QWeb.render('BaseSetting.Tabs',{tabItems : this.modules}));
+        tabs.appendTo(this.$(".settings_tab"));
+
+        $.expr[':'].contains = function(a, i, m) {
+            return jQuery(a).text().toUpperCase()
+                .indexOf(m[3].toUpperCase()) >= 0;
+        };
+
+        this.searchText = "";
+        this.searchInput = this.$('.searchInput');
+        core.bus.on("DOM_updated", this, function() {
+            if (!this.activeTab)
+                this._moveToTab(_.findIndex(this.modules,function(m){return m.key===self.activeSettingTab}));
+        });
+    },
+
+    _getAppIconUrl: function(module) {
+        return module==="generalsettings" ? "/base/static/description/settings.png" : "/"+module+"/static/description/icon.png";
+    },
+
+    _moveToTab: function (index) {
+        this.currentIndex = index === -1 ? 0 : (index === this.modules.length ? index-1 : index);
+        if (this.currentIndex != -1) {
+            if (this.activeView) {
+                this.activeView.addClass("o_hidden");
+            }
+            if (this.activeTab) {
+                this.activeTab.removeClass("selected");
+            }
+            var view = this.modules[this.currentIndex].settingView;
+            var tab = this.$(".tab[data-key='"+this.modules[this.currentIndex].key+"']");
+            view.removeClass("o_hidden");
+            this.activeView = view;
+            this.activeTab = tab;
+
+            if (config.isMobile) {
+                this._activateSettingMobileTab(this.currentIndex);
+            } else {
+                tab.addClass("selected");
+            }
+        }
+    },
+    _activateSettingMobileTab: function (currentTab) {
```

Comment:
What is this supposed to do ?

Reply:
In mobile view behaviour is like swipe content left / right and apps tab will be shown on the top. For animation moving left to right container we have set some properties on each tab and its container. This method will set the required properties (styling and css).